In [18]:
import json
import pandas as pd
import numpy as np

In [26]:
file_dir = "C:\\Users\\sierr\\Data Analytics_Local Documents\\Modules\\Module 8\\Movies-ETL\\Resources\\"

In [27]:
f'{file_dir}wikipedia.movies.json'

'C:\\Users\\sierr\\Data Analytics_Local Documents\\Modules\\Module 8\\Movies-ETL\\Resources\\wikipedia.movies.json'

## Load the JSON into a List of Dictionaries

Using the `with` statement, open the Wikipedia JSON file to be read into the variable `file`, and use `json.load()` to save the data to a new variable.

In [28]:
with open(f'{file_dir}//wikipedia.movies.json', mode='r') as file:
    wiki_movies_raw = json.load(file)

In [29]:
len(wiki_movies_raw)

7311

#### ❓ | PAUSE |


>Is 7,311 a reasonable number of records? We just want to make sure that we don’t have an outlandishly large or small number. If we do, there’s potentially something seriously wrong with the data that needs to be investigated before moving on.
______
>
>One way to check that 7,311 is reasonable is to look at the rate of movies being released that it implies. Rough mental math here is the key—we want these calculations to be quick enough that these checks become a habit. So, let’s say that it’s about 7,200 movies over about 30 years. That’s 240 movies released per year, or a little less than five movies released per week. That seems a little high if we’re considering only major movies, but if for every two major motion pictures, there are three indie films, that doesn’t seem like an outlandish number.

In [30]:
# First 5 records.
wiki_movies_raw[:5]

[{'url': 'https://en.wikipedia.org/wiki/The_Adventures_of_Ford_Fairlane',
  'year': 1990,
  'imdb_link': 'https://www.imdb.com/title/tt0098987/',
  'title': 'The Adventures of Ford Fairlane',
  'Directed by': 'Renny Harlin',
  'Produced by': ['Steve Perry', 'Joel Silver'],
  'Screenplay by': ['David Arnott', 'James Cappe', 'Daniel Waters'],
  'Story by': ['David Arnott', 'James Cappe'],
  'Based on': ['Characters', 'by Rex Weiner'],
  'Starring': ['Andrew Dice Clay',
   'Wayne Newton',
   'Priscilla Presley',
   'Lauren Holly',
   'Morris Day',
   'Robert Englund',
   "Ed O'Neill"],
  'Narrated by': 'Andrew "Dice" Clay',
  'Music by': ['Cliff Eidelman', 'Yello'],
  'Cinematography': 'Oliver Wood',
  'Edited by': 'Michael Tronick',
  'Productioncompany ': 'Silver Pictures',
  'Distributed by': '20th Century Fox',
  'Release date': ['July 11, 1990', '(', '1990-07-11', ')'],
  'Running time': '102 minutes',
  'Country': 'United States',
  'Language': 'English',
  'Budget': '$20 million',


In [31]:
# Last 5 records.
wiki_movies_raw[-5:]

[{'url': 'https://en.wikipedia.org/wiki/Holmes_%26_Watson',
  'year': 2018,
  'imdb_link': 'https://www.imdb.com/title/tt1255919/',
  'title': 'Holmes & Watson',
  'Directed by': 'Etan Cohen',
  'Produced by': ['Will Ferrell',
   'Adam McKay',
   'Jimmy Miller',
   'Clayton Townsend'],
  'Screenplay by': 'Etan Cohen',
  'Based on': ['Sherlock Holmes',
   'and',
   'Dr. Watson',
   'by',
   'Sir Arthur Conan Doyle'],
  'Starring': ['Will Ferrell',
   'John C. Reilly',
   'Rebecca Hall',
   'Rob Brydon',
   'Steve Coogan',
   'Ralph Fiennes'],
  'Music by': 'Mark Mothersbaugh',
  'Cinematography': 'Oliver Wood',
  'Edited by': 'Dean Zimmerman',
  'Productioncompanies ': ['Columbia Pictures',
   'Gary Sanchez Productions',
   'Mosaic Media Group',
   'Mimran Schur Pictures'],
  'Distributed by': 'Sony Pictures Releasing',
  'Release date': ['December 25, 2018',
   '(',
   '2018-12-25',
   ')',
   '(United States)'],
  'Running time': '90 minutes',
  'Country': 'United States',
  'Language

In [32]:
# Some records in the middle.
wiki_movies_raw[3600:3605]

[{'url': 'https://en.wikipedia.org/wiki/Benji:_Off_the_Leash!',
  'year': 2004,
  'imdb_link': 'https://www.imdb.com/title/tt0315273/',
  'title': 'Benji: Off the Leash!',
  'Directed by': 'Joe Camp',
  'Written by': 'Joe Camp',
  'Starring': ['Benji', 'Nick Whitaker', 'Shaggy', 'Gypsy the Cockatoo'],
  'Music by': 'Antonio di Lorenzo',
  'Productioncompany ': 'Mulberry Square Productions',
  'Distributed by': 'Mulberry Square Productions',
  'Release date': ['March 26, 2004', '(', '2004-03-26', ')'],
  'Running time': '97 min',
  'Country': 'United States',
  'Language': 'English',
  'Box office': '$3,817,362'},
 {'url': 'https://en.wikipedia.org/wiki/The_Best_Thief_in_the_World',
  'year': 2004,
  'imdb_link': 'https://www.imdb.com/title/tt0389796/',
  'title': 'The Best Thief in the World',
  'Directed by': 'Jacob Kornbluth',
  'Produced by': ['Tim Perrell', 'Nicola Usborne'],
  'Written by': 'Jacob Kornbluth',
  'Starring': ['Marc Rozendaal',
   'Michael Silverman',
   'David Warsh

In [35]:
kaggle_metadata = pd.read_csv(f'{file_dir}movies_metadata.csv', dtype={'overview': str})
ratings = pd.read_csv(f'{file_dir}ratings.csv')

In [36]:
kaggle_metadata.head()

adult                              belongs_to_collection    budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1  False                                                NaN  65000000   
2  False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3  False                                                NaN  16000000   
4  False  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4                     [{'id': 35, 'name': 'Comedy'}]   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   
3                                   NaN  31357  tt0114885                en   
4                                   NaN  11862  tt0113041                en   

                original_title  \
0                    Toy Story   
1                      Jumanji   
2             Grumpier Old Men   
3            Waiting to Exhale   
4  Father of the Bride Part II   

                                            overview  ... release_date  \
0  Led by Woody, Andy's toys live happily in his ...  ...   1995-10-30   
1  When siblings Judy and Peter discover an encha...  ...   1995-12-15   
2  A family wedding reignites the ancient feud be...  ...   1995-12-22   
3  Cheated on, mistreated and stepped on, the wom...  ...   1995-12-22   
4  Just when George Banks has recovered from his ...  ...   1995-02-10   

       revenue runtime                                   spoken_languages  \
0  373554033.0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0   101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0   127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0   106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title  video vote_average vote_count  
0                    Toy Story  False          7.7     5415.0  
1                      Jumanji  False          6.9     2413.0  
2             Grumpier Old Men  False          6.5       92.0  
3            Waiting to Exhale  False          6.1       34.0  
4  Father of the Bride Part II  False          5.7      173.0  

[5 rows x 24 columns]

In [37]:
kaggle_metadata.tail()

adult belongs_to_collection budget  \
45461  False                   NaN      0   
45462  False                   NaN      0   
45463  False                   NaN      0   
45464  False                   NaN      0   
45465  False                   NaN      0   

                                                  genres  \
45461  [{'id': 18, 'name': 'Drama'}, {'id': 10751, 'n...   
45462                      [{'id': 18, 'name': 'Drama'}]   
45463  [{'id': 28, 'name': 'Action'}, {'id': 18, 'nam...   
45464                                                 []   
45465                                                 []   

                                   homepage      id    imdb_id  \
45461  http://www.imdb.com/title/tt6209470/  439050  tt6209470   
45462                                   NaN  111109  tt2028550   
45463                                   NaN   67758  tt0303758   
45464                                   NaN  227506  tt0008536   
45465                                   NaN  461257  tt6980792   

      original_language       original_title  \
45461                fa              رگ خواب   
45462                tl  Siglo ng Pagluluwal   
45463                en             Betrayal   
45464                en  Satana likuyushchiy   
45465                en             Queerama   

                                                overview  ... release_date  \
45461        Rising and falling between a man and woman.  ...          NaN   
45462  An artist struggles to finish his work while a...  ...   2011-11-17   
45463  When one of her hits goes wrong, a professiona...  ...   2003-08-01   
45464  In a small town live two brothers, one a minis...  ...   1917-10-21   
45465  50 years after decriminalisation of homosexual...  ...   2017-06-09   

      revenue runtime                          spoken_languages    status  \
45461     0.0    90.0    [{'iso_639_1': 'fa', 'name': 'فارسی'}]  Released   
45462     0.0   360.0         [{'iso_639_1': 'tl', 'name': ''}]  Released   
45463     0.0    90.0  [{'iso_639_1': 'en', 'name': 'English'}]  Released   
45464     0.0    87.0                                        []  Released   
45465     0.0    75.0  [{'iso_639_1': 'en', 'name': 'English'}]  Released   

                                          tagline                title  video  \
45461  Rising and falling between a man and woman               Subdue  False   
45462                                         NaN  Century of Birthing  False   
45463                      A deadly game of wits.             Betrayal  False   
45464                                         NaN     Satan Triumphant  False   
45465                                         NaN             Queerama  False   

      vote_average vote_count  
45461          4.0        1.0  
45462          9.0        3.0  
45463          3.8        6.0  
45464          0.0        0.0  
45465          0.0        0.0  

[5 rows x 24 columns]

In [39]:
kaggle_metadata.sample(n=5)

adult belongs_to_collection    budget  \
24972  False                   NaN         0   
18075  False                   NaN         0   
9850   False                   NaN    185000   
44     False                   NaN  20000000   
30918  False                   NaN         0   

                                                  genres homepage      id  \
24972  [{'id': 18, 'name': 'Drama'}, {'id': 35, 'name...      NaN  157380   
18075                      [{'id': 18, 'name': 'Drama'}]      NaN   95448   
9850                      [{'id': 27, 'name': 'Horror'}]      NaN   27495   
44     [{'id': 14, 'name': 'Fantasy'}, {'id': 18, 'na...      NaN     577   
30918                     [{'id': 35, 'name': 'Comedy'}]      NaN  137144   

         imdb_id original_language                     original_title  \
24972  tt2492664                en                  May in the Summer   
18075  tt1781843                fr               L'Exercice de l'État   
9850   tt0078405                en                The Toolbox Murders   
44     tt0114681                en                         To Die For   
30918  tt2461736                en  Jeff Dunham: Minding the Monsters   

                                                overview  ... release_date  \
24972  A bride-to-be is forced to reevaluate her life...  ...   2013-01-17   
18075  Transport Minister Bertrand Saint-Jean is awok...  ...   2011-10-26   
9850   No one is safe when a killer, wielding the dea...  ...   1978-03-15   
44     Susan wants to work in television and will the...  ...   1995-05-20   
30918  Stand-up comedian and ventriloquist Jeff Dunha...  ...   2012-10-07   

          revenue runtime                                   spoken_languages  \
24972         0.0   100.0           [{'iso_639_1': 'en', 'name': 'English'}]   
18075         0.0   115.0  [{'iso_639_1': 'fr', 'name': 'Français'}, {'is...   
9850          0.0    93.0           [{'iso_639_1': 'en', 'name': 'English'}]   
44     21284514.0   106.0           [{'iso_639_1': 'en', 'name': 'English'}]   
30918         0.0    80.0           [{'iso_639_1': 'en', 'name': 'English'}]   

         status                                     tagline  \
24972  Released                  Get ready to clear the air   
18075  Released                                         NaN   
9850   Released  Bit by bit...By bit he carved a nightmare!   
44     Released      All she wanted was a little attention.   
30918  Released                                         NaN   

                                   title  video vote_average vote_count  
24972                  May in the Summer  False          5.8       10.0  
18075                       The Minister  False          5.7       32.0  
9850                 The Toolbox Murders  False          5.4       24.0  
44                            To Die For  False          6.7      177.0  
30918  Jeff Dunham: Minding the Monsters  False          5.6        8.0  

[5 rows x 24 columns]

In [40]:
ratings.head()

userId  movieId  rating   timestamp
0       1      110     1.0  1425941529
1       1      147     4.5  1425942435
2       1      858     5.0  1425941523
3       1     1221     5.0  1425941546
4       1     1246     5.0  1425941556

In [41]:
ratings.tail()

userId  movieId  rating   timestamp
26024284  270896    58559     5.0  1257031564
26024285  270896    60069     5.0  1257032032
26024286  270896    63082     4.5  1257031764
26024287  270896    64957     4.5  1257033990
26024288  270896    71878     2.0  1257031858

In [42]:
ratings.sample(n=5)

userId  movieId  rating   timestamp
1697607    17667     4447     3.5  1255279540
5359582    55273     3911     3.0   973365796
4457895    45811   175309     3.5  1501022644
16403399  170505      785     2.5  1094012691
21653984  224925     1240     3.0  1365638945